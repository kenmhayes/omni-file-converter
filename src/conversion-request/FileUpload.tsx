import React from 'react';
import { FileUploader } from 'react-drag-drop-files';
import FileList from './FileList';

export interface FileUploadProps {
  disabled?: boolean;
  fileTypes: string[];
  files: File[];
  setFiles: (files: File[]) => void;
}

function FileUpload(props: FileUploadProps) {
  const {
    fileTypes, files, setFiles, disabled,
  } = props;

  const handleChange = (file: File) => {
    const newFiles = [...files];
    newFiles.push(file);
    setFiles(newFiles);
  };

  return (
    <div>
      <FileUploader
        disabled={disabled}
        handleChange={handleChange}
        name="file"
        files={files}
        types={fileTypes}
      />
      <FileList files={files} setFiles={setFiles} />
    </div>
  );
}

FileUpload.defaultProps = {
  disabled: false,
};

export default FileUpload;
