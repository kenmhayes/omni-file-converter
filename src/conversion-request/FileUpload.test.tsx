import React from 'react';
import { create } from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import FileUpload from './FileUpload';

it('renders a file list', () => {
  const files: File[] = [
    { name: "File 1" } as File, { name: "File 2" } as File, { name: "File 3"} as File
  ];
  const fileTypes = [ "png" ];
  const component = <FileUpload files={files} setFiles={jest.fn()} fileTypes={fileTypes} />;

  expect(create(component).toJSON()).toMatchSnapshot();
});

it('changes file list on upload and removal', async () => {
    const setFiles = jest.fn();
    const files: File[] = [
        { name: "File 1" } as File, { name: "File 2" } as File, { name: "File 3"} as File
      ];
    const fileTypes = [ "png" ];
    const newFile = new File(['hello'], 'hello.png', {type: 'image/png'})

    const component = <FileUpload files={files} setFiles={setFiles} fileTypes={fileTypes} />;
    render(component);

    // Find file input, doesn't seem possible with react testing library with this 3rd party component
    const fileInputQuery = document.querySelector("input");
    expect(fileInputQuery).toBeTruthy();
    const fileInput = fileInputQuery as HTMLInputElement;

    fireEvent.change(fileInput, { target: { files: [newFile] }});

    expect(setFiles).toHaveBeenCalledWith([...files, newFile]);
});
