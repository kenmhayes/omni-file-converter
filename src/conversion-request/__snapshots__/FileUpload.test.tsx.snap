// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a file list 1`] = `
<div>
  <label
    className="sc-aXZVg guZdik  "
    htmlFor="file"
    onClick={[Function]}
  >
    <input
      accept=".png"
      disabled={false}
      name="file"
      onChange={[Function]}
      onClick={[Function]}
      type="file"
    />
    <svg
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.33317 6.66667H22.6665V16H25.3332V6.66667C25.3332 5.196 24.1372 4 22.6665 4H5.33317C3.8625 4 2.6665 5.196 2.6665 6.66667V22.6667C2.6665 24.1373 3.8625 25.3333 5.33317 25.3333H15.9998V22.6667H5.33317V6.66667Z"
        fill="#0658C2"
      />
      <path
        d="M10.6665 14.6667L6.6665 20H21.3332L15.9998 12L11.9998 17.3333L10.6665 14.6667Z"
        fill="#0658C2"
      />
      <path
        d="M25.3332 18.6667H22.6665V22.6667H18.6665V25.3333H22.6665V29.3333H25.3332V25.3333H29.3332V22.6667H25.3332V18.6667Z"
        fill="#0658C2"
      />
    </svg>
    <div
      className="sc-eqUAAy jWkLDY"
    >
      <span
        className="sc-fqkvVR kFhUBM"
      >
        <span>
          Upload
        </span>
         or drop a file right here
      </span>
      <span
        className="file-types"
        title="types: png"
      >
        png
      </span>
    </div>
  </label>
  <div
    className="container"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-9"
        >
          File 1
        </div>
        <div
          className="col"
        >
          <button
            className="btn"
            onClick={[Function]}
            title="Remove File"
            type="button"
          >
            <img
              alt="Remove File"
              height="32"
              src="x-circle.svg"
              width="32"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-9"
        >
          File 2
        </div>
        <div
          className="col"
        >
          <button
            className="btn"
            onClick={[Function]}
            title="Remove File"
            type="button"
          >
            <img
              alt="Remove File"
              height="32"
              src="x-circle.svg"
              width="32"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-9"
        >
          File 3
        </div>
        <div
          className="col"
        >
          <button
            className="btn"
            onClick={[Function]}
            title="Remove File"
            type="button"
          >
            <img
              alt="Remove File"
              height="32"
              src="x-circle.svg"
              width="32"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
